web_poet/example.py

```diff
@@ -0,0 +1,100 @@
+"""This is a simplified, incomplete example of a web-poet framework, written as
+support material for the web-poet tutorial.
+
+No part of this module is intended for production use, and it may change in a
+backward-incompatible way at any point in the future.
+"""
+
+from asyncio import run
+from typing import Any, Dict, Optional, Type
+from warnings import warn
+
+import andi
+import requests
+from url_matcher import URLMatcher
+
+from . import default_registry
+from .page_inputs import HttpClient, HttpResponse, PageParams
+from .pages import ItemPage, is_injectable
+from .utils import ensure_awaitable
+
+warn(
+    (
+        "You should only be importing web_poet.example to follow the web-poet "
+        "tutorial, never as part of production code."
+    ),
+    UserWarning,
+    stacklevel=2,
+)
+
+
+class _HttpClient:
+    async def get(self, url: str) -> HttpResponse:
+        return _get_http_response(url)
+
+
+def _get_page_class(url: str, item_class: Type) -> Type[ItemPage]:
+    url_matcher = URLMatcher(
+        {
+            rule.use: rule.for_patterns
+            for rule in default_registry.get_rules()
+            if rule.to_return == item_class
```

Comment:
We have a new `.page_object_for_item()` method in https://github.com/scrapinghub/web-poet/pull/112/files#diff-22f478b54a07b372ea47ead887ca01488362ad501692903a9d303511f9d1fe24R383-R394